tensorflow/core/kernels/mkl_tfconv_op.cc

```diff
@@ -29,8 +29,8 @@ limitations under the License.
 #include "tensorflow/core/util/tensor_format.h"
 
 #include "tensorflow/core/util/mkl_util.h"
-#include "third_party/mkl/include/mkl_dnn.h"
-#include "third_party/mkl/include/mkl_dnn_types.h"
+#include "mkl_dnn.h"
```

Comment:
That convention is for internal includes. Anything that is a part of tensorflow. If something is an external dependency, we point our build rules to their include folder:
https://github.com/tensorflow/tensorflow/blob/master/third_party/png.BUILD#L31

Then include them directly in our files. This is what I am doing here. Getting rid of brittle downloading through configure script mechanism end up moving us to here.
Unless we check in the headers into our third_party/mkl folder, we cannot make them accessible through the `third_party/mkl/include`path.

Moving everything into a single header is something I had in mind, but that will come in a later PR.